dataikuapi/dss/ml.py

```diff
@@ -245,13 +248,25 @@ def get_algorithm_settings(self, algorithm_name):
         Please refer to the documentation for details on available algorithms.
 
         :param str algorithm_name: Name (in capitals) of the algorithm.
-        :return: A dict of the settings for an algorithm
-        :rtype: dict 
-        """
-        if algorithm_name in self.__class__.algorithm_remap:
-            algorithm_name = self.__class__.algorithm_remap[algorithm_name]
-
-        return self.mltask_settings["modeling"][algorithm_name.lower()]
+        :return: An AlgortihmSettings (extended dict) for an algorithm.
+        :rtype: AlgorithmSettings
+        """
+        if algorithm_name in ["CUSTOM_MLLIB", "CUSTOM_PYTHON", "PLUGIN_PYTHON"]:
+            return self.mltask_settings["modeling"][algorithm_name.lower()]
+        elif algorithm_name in self.__class__.algorithm_remap:
+            algorithm_meta = self.__class__.algorithm_remap[algorithm_name]
+            algorithm_name = algorithm_meta.algorithm_name
+            algorithm_settings_class = algorithm_meta.algorithm_settings_class
```

Comment:
Raising the error for unknown algorithm names is still relevant

```suggestion
            algorithm_settings_class = algorithm_meta.algorithm_settings_class
        else:
            raise ValueError("Unknown algorithm: {}".format(algorithm_name))	
```